### ISSUE TITLE
H test failures

### ISSUE THREAD

**Issue by bergmark** [tweag/HaskellR#259]
Since H is back in stackage...

The test-suite for this package has been disabled in stackage because of this issue. If a fix is published to hackage could you please send a PR to re-enable the tests to stackage (it's in the expected-test-failures section) or ping me? If some new libraries need to be installed on the stackage build server please send a pull request to debian-bootstrap.sh.

Thanks!

```
> /tmp/stackage-build8/H-0.9.0.0$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup configure --enable-tests --package-db=clear --package-db=global --package-db=/var/stackage/work/builds/nightly/pkgdb --libdir=/var/stackage/work/builds/nightly/lib --bindir=/var/stackage/work/builds/nightly/bin --datadir=/var/stackage/work/builds/nightly/share --libexecdir=/var/stackage/work/builds/nightly/libexec --sysconfdir=/var/stackage/work/builds/nightly/etc --docdir=/var/stackage/work/builds/nightly/doc/H-0.9.0.0 --htmldir=/var/stackage/work/builds/nightly/doc/H-0.9.0.0 --haddockdir=/var/stackage/work/builds/nightly/doc/H-0.9.0.0 --flags=
Configuring H-0.9.0.0...
> /tmp/stackage-build8/H-0.9.0.0$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup build
Building H-0.9.0.0...
Preprocessing executable 'H' for H-0.9.0.0...
[1 of 2] Compiling Paths_H          ( dist/build/autogen/Paths_H.hs, dist/build/H/H-tmp/Paths_H.o )
[2 of 2] Compiling Main             ( H.hs, dist/build/H/H-tmp/Main.o )
Linking dist/build/H/H ...
Preprocessing test suite 'tests' for H-0.9.0.0...
[1 of 1] Compiling Main             ( tests/tests.hs, dist/build/tests/tests-tmp/Main.o )
Linking dist/build/tests/tests ...
> /tmp/stackage-build8/H-0.9.0.0$ dist/build/tests/tests
Integration tests
  qq: FAIL (0.50s)
    Outputs don't match.
    Expected:
    [1] 1
    [1] 3
    [1] "1" "2" "3"
    [1] 2
    [1] 3
    [1] 6
    [1] 8
     [1]  1  2  3  4  5  6  7  8  9 10
    [1] 3
     [1]  2  3  4  5  6  7  8  9 10 11
    [1] 43
    [1] 1 2 3
    [1] 8
    NULL
    [1] 3
    [1] 99
    [1] 120
    [1] 29
    blah
    class generator function for class "x-test" from package '.GlobalEnv'
    function (...) 
    new("x-test", ...)
    An object of class "x-test"
    Slot "a":
    [1] 1
    
    Slot "b":
    [1] 2
    
    An object of class "x-test"
    Slot "a":
    [1] 1
    
    Slot "b":
    [1] 2
    
    [1] 1
    [1] 2
    [1] "S4"
    [1] "S4"
    An object of class "x-test"
    Slot "a":
    [1] 1
    
    Slot "b":
    [1] 2
    
    [1] "S4"
    
    
    Actual:

1 out of 1 tests failed (0.50s)
>>>

**Comment by qnikst** [comment]
Hi, @bergmark.
I've changed my workflow and reproduced a error. Using `stack test` works fine, but if you run executable manually you need certain precondition for test to run, like: current working directory should be root of the `H` project and `H` executable should be in `$PATH` and possibly few other that I'm trying to investigate.
I'll try to make this kind of workflow possible and update tests.

**Comment by qnikst** [comment]
Just realized that test should be running from it's sandbox so `ghci` that is inside `H` could load libraries. Is it possible to modify test execution to support that or should I run `H` via `stack exec` from the test itself?

I don't like latter because `stack`-less workflow will not be possible in that case, but afaiu it's not officially supported anyway.

**Comment by bergmark** [comment]
The working directory will be what you want, but the stackage build server doesn't use stack to build packages. What you describe might work, but I'm not sure. We can always try.

**Comment by qnikst** [comment]
Oh! For some reason I have missed that.  In this case setting  `GHC_PACKAGE_PATH="/var/stackage/work/builds/nightly/pkgdb"` should fix tests. But I'm not sure how to make it automatic. I will try to search a way that will make tests work under different workflows.

**Comment by qnikst** [comment]
@bergmark those tests were dropped as they are not useful, once new version will out expected-tests-failures section will be removed, but AFAIU there is no hurry because for stackage users nothing will change? I'm keeping this issue open until we will not release next version.

**Comment by mboes** [comment]
Fixed in latest release of H: v0.9.0.1.
